lib/translate/inquiry.cc

```diff
@@ -0,0 +1,154 @@
+        result.peripheral_qualifier = static_cast<scsi_defs::PeripheralQualifier>(0);
+        result.peripheral_device_type = static_cast<scsi_defs::PeripheralDeviceType>(0);
+        result.rmb = 0;
+        result.version = static_cast<scsi_defs::Version>(0x6);
+        result.normaca = 0;
+        result.hisup = 0;
+        result.response_data_format = static_cast<scsi_defs::ResponseDataFormat>(0b10);
+        result.additional_length = 0x1f;
+        result.sccs = 0;
+        result.acc = 0;
+        result.tpgs = static_cast<scsi_defs::TPGS>(0);
+        result.third_party_copy = 0;
+        result.protect = (identify_namespace_data.dps.pit == 0 && identify_namespace_data.dps.md_start == 0) ? 0 : 1;
+        result.encserv = 0;
+        result.multip = 0;
+        result.addr_16 = 0;
+        result.wbus_16 = 0;
+        result.sync = 0;
+        result.cmdque = 1;
+
+        // Shall be set to “NVMe” followed by 4 spaces: “NVMe “
+        char NVME_VENDOR_IDENTIFICATION[9] = "NVMe    ";
+        for(int i = 0; i < 8; i++) {
+            result.vendor_identification <<= 8;
+            result.vendor_identification |= NVME_VENDOR_IDENTIFICATION[i];
+        }
+        // printf("%lux\n", result.vendor_identification);
+
+        // Shall be set to the first 16 bytes of the Model Number (MN) field within the Identify Controller Data Structure
+        for(int i = 0; i < 16; i++) {
+            result.product_identification[i] = identify_controller_data.mn[i];
+        }
+
+        // Shall be set to the first 4 bytes of the Firmware Revision (FR) field within the Identify Controller Data Structure
+        for(int i = 0; i < 4; i++) {
+            result.product_revision_level <<= 8;
+            result.product_revision_level |= identify_controller_data.fr[i];
+        }
+        return result;
+    }
+    scsi_defs::SupportedVitalProductDataPages build_supported_vpd_pages_data()
+    {
+        scsi_defs::SupportedVitalProductDataPages result = scsi_defs::SupportedVitalProductDataPages();
+        result.peripheral_qualifier = scsi_defs::PeripheralQualifier::kPeripheralDeviceConnected;
+        result.peripheral_device_type = scsi_defs::PeripheralDeviceType::kDirectAccessBlock;
+
+        // Shall be set to 5 indicating number of items supported VPD pages list requires.
+        // NOTE: document says to set this to 5 but there are 7 entries....
+        result.page_length = 5;
+
+        result.supported_page_list[0] = 0x00;
+        result.supported_page_list[1] = 0x80;
+        result.supported_page_list[2] = 0x83;
+        result.supported_page_list[3] = 0x86;
+        result.supported_page_list[4] = 0xB0;
+        result.supported_page_list[5] = 0xB1;
+        result.supported_page_list[6] = 0xB2;
+        return result;
+    }
+
+    // TODO: figure out return value...
+    // Main logic engine for the Inquiry command
+    void translate(uint32_t* raw_cmd, int buffer_length) {
```

Comment:
This should take an `absl::Span<const uint32_t>`, assuming `buffer_length` is a number of `uint32_t`s.